src/clarity_vm/clarity.rs

```diff
@@ -105,13 +105,18 @@ pub struct ClarityInstance {
     mainnet: bool,
 }
 
+pub struct PreCommitClarityBlock<'a> {
```

Comment:
Can you document why this struct is necessary?  i.e. we need to do other blocking I/O (such as post events to an event dispatcher) before committing to the chainstate headers MARF and the Clarity DB MARF, so we need to effectively "seal" an ongoing Clarity transaction into this struct so we can do the blocking I/O first (instead of in-between these two transactions)?

Reply:
Added rustdocs for this struct in 8608cc7fd